src/contributing.md

```diff
@@ -0,0 +1,140 @@
+# Contributing guidelines
+
+## Generic Reusable Component Design
+
+Define the component's purpose. Narrow down the scope as much as possible.
+
+Avoid assumptions about its parent/owner.
+
+Define the interface/contract. Make explicit what characteristics can be set by the owner and which are read-only or undefined behavior. (prefix with `_` private internal properties?)
```

Comment:
Since we never did the `_` prefix before, maybe it's not worth adding here.

Reply:
Agree. I changed to state the importance of having a clear interface.